src/scanners/binary.js

```diff
@@ -13,7 +13,9 @@ export default class BinaryScanner extends BaseScanner {
   }
 
   check(buffer, values) {
-    for (let v in values) {
+    const valuesKeys = Object.keys(values);
+    for (let i = 0; i < valuesKeys.length; i++) {
+      const v = valuesKeys[i];
```

Comment:
Because there's no way to break out of a `forEach` without throwing. 

Thinking about it this is probably something where we could be using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

It could probably be re-written as:

```
if (Object.keys(values).some((v) => values[v] !== buffer[v])) {
  return;
}
```